stl/inc/xutility

```diff
@@ -1859,32 +1904,55 @@ public:
         return _Tmp;
     }
 
-    // N.B. functions valid for random-access iterators only beyond this point
+    _NODISCARD _CONSTEXPR17 reverse_iterator operator+(const difference_type _Off) const {
+        return reverse_iterator(current - _Off);
+    }
 
     _CONSTEXPR17 reverse_iterator& operator+=(const difference_type _Off) {
         current -= _Off;
         return *this;
     }
 
-    _NODISCARD _CONSTEXPR17 reverse_iterator operator+(const difference_type _Off) const {
-        return reverse_iterator(current - _Off);
+    _NODISCARD _CONSTEXPR17 reverse_iterator operator-(const difference_type _Off) const {
+        return reverse_iterator(current + _Off);
     }
 
     _CONSTEXPR17 reverse_iterator& operator-=(const difference_type _Off) {
         current += _Off;
         return *this;
     }
 
-    _NODISCARD _CONSTEXPR17 reverse_iterator operator-(const difference_type _Off) const {
-        return reverse_iterator(current + _Off);
-    }
-
     _NODISCARD _CONSTEXPR17 reference operator[](const difference_type _Off) const {
         return current[static_cast<difference_type>(-_Off - 1)];
     }
 
+#ifdef __cpp_lib_concepts
+    _NODISCARD friend constexpr iter_rvalue_reference_t<_BidIt> iter_move(const reverse_iterator& _It) noexcept(
+        is_nothrow_copy_constructible_v<_BidIt>&& noexcept(_RANGES iter_move(--_STD declval<_BidIt&>()))) {
+        auto _Tmp = _It.current;
+        --_Tmp;
+        return _RANGES iter_move(_Tmp);
+    }
+
+    template <indirectly_swappable<_BidIt> _BidIt2>
+    friend constexpr void iter_swap(const reverse_iterator& _Left, const reverse_iterator<_BidIt2>& _Right) noexcept(
+        is_nothrow_copy_constructible_v<_BidIt>&& is_nothrow_copy_constructible_v<_BidIt2>&& noexcept(
+            _RANGES iter_swap(--_STD declval<_BidIt&>(), --_STD declval<_BidIt2&>()))) {
+        auto _LTmp = _Left.current;
+        auto _RTmp = _Right.base();

```

Comment:
Can this be `_Right.current` because `reverse_iterator` grants friendship to other `reverse_iterator`s now, or does that not work in these friend definitions?